src/Microsoft.ApplicationInsights/Extensibility/Implementation/ApplicationId/ApplicationIdHelper.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.ApplicationId
+{
+    using System.Globalization;
+    using System.Text.RegularExpressions;
+
+    internal static class ApplicationIdHelper
+    {
+        /// <summary>
+        /// special string which describes that ID was taken from Breeze
+        /// </summary>
+        private const string Format = "cid-v1:{0}";
+
+        /// <summary>
+        /// Max length of Application Id allowed in response from Breeze.
+        /// </summary>
+        private const int ApplicationIdMaxLengeth = 50;
+
+        /// <summary>
+        /// Format an Application Id string (ex: 00000000-0000-0000-0000-000000000000) 
+        /// as (ex: cid-v1:00000000-0000-0000-0000-000000000000).
+        /// </summary>
+        /// <param name="applicationId">Application Id is expected to be a Guid string.</param>
+        /// <remarks>
+        /// To protect against injection attacks, Application Id will be truncated to a maximum length.
+        /// Application Ids are expected to Http Header safe, and all non-ASCII characters will be removed.
+        /// </remarks>
+        internal static string ApplyFormatting(string applicationId)
+        {
+            // Arbitrary maximum length to guard against injections.
+            applicationId = EnforceMaxLength(applicationId, ApplicationIdMaxLengeth);
+            if (string.IsNullOrWhiteSpace(applicationId))
+            {
+                return null;
+            }
+
+            // String must be sanitized to include only characters safe for http header.
+            applicationId = SanitizeString(applicationId);
+
+            return string.Format(CultureInfo.InvariantCulture, Format, applicationId);
+        }
+
+        /// <summary>
+        /// Check a strings length and trim to a max length if needed.
+        /// </summary>
+        private static string EnforceMaxLength(string input, int maxLength)
+        {
+            if (input != null && input.Length > maxLength)
+            {
+                input = input.Substring(0, maxLength);
+            }
+
+            return input;
+        }
+
+        /// <summary>
+        /// Application Id will eventually end up in Http Headers, which only allow Printable US-ASCII characters.
+        /// Remove all other characters.
+        /// </summary>
+        private static string SanitizeString(string input)
+        {
+            if (string.IsNullOrWhiteSpace(input))
+            {
+                return input;
+            }
+
+            // US-ASCII characters (hex: 0x00 - 0x7F) (decimal: 0-127)
+            // ASCII Extended characters (hex: 0x80 - 0xFF) (decimal: 0-255) (NOT ALLOWED)
+            // Non-Printable ASCII characters are (hex: 0x00 - 0x1F) (decimal: 0-31) (NOT ALLOWED)
+            // Printable ASCII characters are (hex: 0x20 - 0xFF) (decimal: 32-255) 
+            return Regex.Replace(input, @"[^\u0020-\u007F]", string.Empty);
```

Comment:
👎 for using a Regex here -- considering that we expect replacement to be very rare because the input is almost certainly a GUID.

Reply:
that addition came from a bug fix. (i'll try to find the link)
if you have suggestions for an alternative I'd love to hear it but as of now that check is very necessary.

Reply:
Frankly, even if we wanted to change from a GUID, I don't think we could. And even then, would we really use something that used non-printable characters?

I suggest:

Avoid the Regex unless you can't. i.e. First test to see if all chars are 'sane' and only sanitize if/when you find one.

If the rule is "any char outside the 20-7F range should be removed", then:
```c#
private static bool IsSane(char ch) => (uint)(ch - 0x20) <= (0x7F - 0x20);

private static string SanitizeString(string input)
{
    if (input == null)
    {
        return null;
    }

    foreach(var ch in input)
    {
        if (!IsSane(ch))
        {
            return Regex.Replace(input, @"[etc...
        }
    }

    return input;
}
```